Src/Couchbase.Linq/Extensions/BucketExtensions.cs

```diff
@@ -8,7 +9,8 @@ public static class BucketExtensions
     {
         public static IQueryable<T> Queryable<T>(this IBucket bucket)
         {
-            return EntityFilterManager.ApplyFilters(new BucketQueryable<T>(bucket));
+            //TODO refactor so ClientConfiguration is injectable
+            return EntityFilterManager.ApplyFilters(new BucketQueryable<T>(bucket, new ClientConfiguration()));
```

Comment:
@brantburnett -

This class is now internal and then only place I see it being used was in the tests. The public interface here is now DBContext (Btw, shouldn't this be BucketContext?). I was going to add another param to Queryable so that it was injectible, but figured we could do that later...the LOE is the changes to 184 tests that depend on that method.

The goal is to make the ICluster implementation the dependency from  the provider, and remove the dependency on ClusterHelper - which can still be used at the application level. The ICluster object has a ClientConfiguration field which contains the necessary dependencies for the name resolver and translator. 

I agree about exposing the configuration info from IBucket, but it requires a change to couchbase-net-client, which would require a release before we could use it here. I think a better tactic is to do that in a later release.

-Jeff